docs/orleans/grains/transactions.md

```diff
@@ -115,54 +97,66 @@ public class TransactionalStateAttribute : Attribute
+
+## Call transaction methods from another grain
+
+Transactional methods on a grain interface are called like any other grain method. As an alternative approach using the `ITransactionClient`, the `AtmGrain` implementation below calls the `Transfer` method (which is transactional) on the `IAccountGrain` interface.
+
+Consider the `AtmGrain` implementation, which resolves the two referenced account grains and makes the appropriate calls to `Withdraw` and `Deposit`:
+
+:::code source="snippets/transactions/Grains/AtmGrain.cs":::
+
+Your client app code can call `AtmGrain.Transfer` in a transactional manner as follows:
 
 ```csharp
-IATMGrain atm = client.GetGrain<IATMGrain>(0);
+IAtmGrain atmOne = client.GetGrain<IAtmGrain>(0);
+
 Guid from = Guid.NewGuid();
 Guid to = Guid.NewGuid();
-await atm.Transfer(from, to, 100);
+
+await atmOne.Transfer(from, to, 100);
+
 uint fromBalance = await client.GetGrain<IAccountGrain>(from).GetBalance();
 uint toBalance = await client.GetGrain<IAccountGrain>(to).GetBalance();
 ```
 
-In the above calls, an ATM grain is used to transfer 100 units of currency from one account to another. After the transfer is complete, both accounts are queried to get their current balance. The currency transfer, as well as both account queries, are performed as ACID transactions.
+In the above calls, an `IAtmGrain` is used to transfer 100 units of currency from one account to another. After the transfer is complete, both accounts are queried to get their current balance. The currency transfer, as well as both account queries, are performed as ACID transactions.
 
-As seen in the above example, transactions can return values within a task, like other grain calls, but upon call failure, they will not throw application exceptions, but rather an <xref:Orleans.Transactions.OrleansTransactionException> or <xref:System.TimeoutException>. If the application throws an exception during the transaction and that exception causes the transaction to fail (as opposed to failing because of other system failures), the application exception will be the inner exception of the `OrleansTransactionException`.
+As shown in the preceding example, transactions can return values within a `Task`, like other grain calls, but upon call failure, they will not throw application exceptions, but rather an <xref:Orleans.Transactions.OrleansTransactionException> or <xref:System.TimeoutException>. If the application throws an exception during the transaction and that exception causes the transaction to fail (as opposed to failing because of other system failures), the application exception will be the inner exception of the `OrleansTransactionException`.
```

Comment:
```suggestion
As shown in the preceding example, transactions can return values within a `Task`, like other grain calls. But upon call failure, they will not throw application exceptions but rather an <xref:Orleans.Transactions.OrleansTransactionException> or <xref:System.TimeoutException>. If the application throws an exception during the transaction and that exception causes the transaction to fail (as opposed to failing because of other system failures), the application exception will be the inner exception of the `OrleansTransactionException`.
```